From 2de9327cf0cb117b9411e7fb83fa604827f5aa2f Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 13 Apr 2026 07:05:10 +0000
Subject: [PATCH] of: unittest: Fix of_count_phandle_with_args() expected value
 message

jira VULN-35422
cve-bf CVE-2023-52679
commit-author Geert Uytterhoeven <geert+renesas@glider.be>
commit 716089b417cf98d01f0dc1b39f9c47e1d7b4c965

The expected result value for the call to of_count_phandle_with_args()
was updated from 7 to 8, but the accompanying error message was
forgotten.

Fixes: 4dde83569832f937 ("of: Fix double free in of_parse_phandle_with_args_map")
	Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
Link: https://lore.kernel.org/r/20240111085025.2073894-1-geert+renesas@glider.be
	Signed-off-by: Rob Herring <robh@kernel.org>
(cherry picked from commit 716089b417cf98d01f0dc1b39f9c47e1d7b4c965)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 drivers/of/unittest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/unittest.c b/drivers/of/unittest.c
index eb6b3ab51ce8..486cc471c20c 100644
--- a/drivers/of/unittest.c
+++ b/drivers/of/unittest.c
@@ -572,7 +572,7 @@ static void __init of_unittest_parse_phandle_with_args_map(void)
 	}
 
 	rc = of_count_phandle_with_args(np, "phandle-list", "#phandle-cells");
-	unittest(rc == 8, "of_count_phandle_with_args() returned %i, expected 7\n", rc);
+	unittest(rc == 8, "of_count_phandle_with_args() returned %i, expected 8\n", rc);
 
 	for (i = 0; i < 9; i++) {
 		bool passed = true;
-- 
2.47.3

